hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -3146,3 +3173,70 @@ func globalConfigString(hcluster *hyperv1.HostedCluster) (string, error) {
 func payloadConfigHash(config, targetVersion, pullSecretName, globalConfig string) string {
 	return supportutil.HashSimple(config + targetVersion + pullSecretName + globalConfig)
 }
+
+// mirrorConfigs mirrors higher-level machineConfiguration objects, i.e., kubeletConfig and containerRuntimeConfig, into
+// the HCP namespace, since those are needed as an input for certain operators (such as NTO)
+func (r *NodePoolReconciler) mirrorConfigs(ctx context.Context, logr logr.Logger, mirroredConfigs []corev1.ConfigMap, controlPlaneNamespace string, nodePool *hyperv1.NodePool) error {
+	// get kubeletConfig objects from the HCP namespace
+	kubeletConfigConfigMapList := &corev1.ConfigMapList{}
+	if err := r.List(ctx, kubeletConfigConfigMapList, &client.ListOptions{
+		Namespace:     controlPlaneNamespace,
+		LabelSelector: labels.SelectorFromValidatedSet(labels.Set{mirroredConfigLabel: ""}),
+	}); err != nil && !apierrors.IsNotFound(err) {
+		return err
+	}
+	// get containerRuntimeConfig objects from the HCP namespace
+	containerRuntimeConfigConfigMapList := &corev1.ConfigMapList{}
+	if err := r.List(ctx, containerRuntimeConfigConfigMapList, &client.ListOptions{
+		Namespace:     controlPlaneNamespace,
+		LabelSelector: labels.SelectorFromValidatedSet(labels.Set{mirroredConfigLabel: ""}),
+	}); err != nil && !apierrors.IsNotFound(err) {
+		return err
+	}
+	existingConfigs := append(kubeletConfigConfigMapList.Items, containerRuntimeConfigConfigMapList.Items...)
```

Comment:
If any action needs to be taken, please emit a log here using `sets.SymmetricDifference` to explain what action needs to be taken.

Reply:
Removed, no need.